plugins/woocommerce/client/blocks/assets/js/extensions/payment-methods/paypal/buttons.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import { PayPalScriptProvider, PayPalButtons } from '@paypal/react-paypal-js';
+import { getPaymentMethodData } from '@woocommerce/settings';
+import { dispatch } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * PayPalButtonsContainer component.
+ *
+ * @param {Object}  props
+ * @param {string}  props.clientId
+ * @param {string}  [props.components]
+ * @param {string}  [props.disableFunding]
+ * @param {string}  [props.enableFunding]
+ * @param {string}  [props.currency]
+ * @param {string}  [props.intent]
+ * @param {string}  [props.merchantId]
+ * @param {string}  [props.partnerAttributionId]
+ * @param {string}  [props.pageType]
+ * @param {boolean} [props.isProductPage]
+ * @return {JSX.Element} The PayPal Buttons container component.
+ */
+const PayPalButtonsContainer = ( {
+	clientId,
+	components,
+	disableFunding,
+	enableFunding,
+	currency,
+	intent,
+	merchantId,
+	partnerAttributionId,
+	pageType,
+	isProductPage,
+} ) => {
+	const [ orderReceivedUrl, setOrderReceivedURL ] = useState( '' );
+	const [ orderId, setOrderId ] = useState( '' );
+	const payPalData = getPaymentMethodData( 'paypal', {} );
+	const options = {
+		clientId: clientId || '',
+		components: components || '',
+		disableFunding: disableFunding || '',
+		enableFunding: enableFunding || '',
+		currency: currency || '',
+		intent: intent || '',
+		merchantId: merchantId || '',
+		'data-partner-attribution-id': partnerAttributionId || '',
+		'data-page-type': pageType || '',
+	};
+
+	const createOrder = async ( data ) => {
+		let responseData;
+		try {
+			// If we're inside the product page, we need to empty the cart,
+			// and add the current product to the cart.
+			if ( isProductPage ) {
```

Comment:
I think there's opportunities in here to use the existing cart and checkout data stores.  See https://github.com/woocommerce/woocommerce/blob/trunk/docs/block-development/reference/data-store/cart.md